src/os/bluestore/BlueStore.cc

```diff
@@ -8666,21 +8664,15 @@ void BlueStore::_deferred_submit_unlock(OpSequencer *osr)
 void BlueStore::_deferred_aio_finish(OpSequencer *osr)
 {
   dout(10) << __func__ << " osr " << osr << dendl;
-  assert(osr->deferred_running);
-  DeferredBatch *b = osr->deferred_running;
+  DeferredBatch *b;
 
   {
     std::lock_guard<std::mutex> l(deferred_lock);
-    assert(osr->deferred_running == b);
-    osr->deferred_running = nullptr;
-    if (!osr->deferred_pending) {
+    b = osr->deferred_running.front();
```

Comment:
If we allow multiple DeferredBatches in flight, there seems to be no guarantee this will be the exact one that triggers `aio_finish()` here?